File: Modules/ServerTypes/Ticket/TimeFlagColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TimeFlagFormatter } from "../../Ticket/Priority/ColorPickerEditor";
import { TimeFlagRow } from "./TimeFlagRow";

export interface TimeFlagColumns {
    Id: Column<TimeFlagRow>;
    Name: Column<TimeFlagRow>;
    DuePercent: Column<TimeFlagRow>;
    Color: Column<TimeFlagRow>;
    Description: Column<TimeFlagRow>;
}

export class TimeFlagColumns extends ColumnsBase<TimeFlagRow> {
    static readonly columnsKey = 'Ticket.TimeFlag';
    static readonly Fields = fieldsProxy<TimeFlagColumns>();
}

[TimeFlagFormatter]; // referenced types

File: Modules/Ticket/TimeFlag/TimeFlagColumns.cs
namespace Ticketing.Ticket.Columns;

[ColumnsScript("Ticket.TimeFlag")]
[BasedOnRow(typeof(TimeFlagRow), CheckNames = true)]
public class TimeFlagColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public int DuePercent { get; set; }
    public string IconName { get; set; }
    public string Color { get; set; }
    public string Description { get; set; }
}